base/compiler/ssair/inlining.jl

```diff
@@ -925,8 +923,9 @@ function call_sig(ir::IRCode, stmt::Expr)
     Signature(f, ft, argtypes)
 end
 
-function inline_apply!(ir::IRCode, todo::Vector{Pair{Int, Any}}, idx::Int, sig::Signature,
-                       state::InliningState)
+function inline_apply!(
+    ir::IRCode, idx::Int, stmt::Expr, sig::Signature,
+    state::InliningState, todo::Vector{Pair{Int, Any}})
     stmt = ir.stmts[idx][:inst]
```

Comment:
Though I think we should move towards passing around `ir.stmt[idx]` objects, instead of unpacking them early everywhere
```suggestion
```

Reply:
@vtjnash Do you think fc033c008b3324dcf47aa2b98c904187f37d7836 is a good refactoring?

Reply:
I don't really like having an expensive and tedious copy operation instead of using a lightweight Instruction directly to access the information directly, when possible

Reply:
Hm, rather I thought having `InstInfo` passed around would reduce the cost of array indexing that otherwise happens at many places to unpack information. But creating `InstInfo`s would defeat this purpose by introducing high overheads involved with those allocations?